src/main.rs

```diff
@@ -40,22 +39,56 @@ fn get_user_agent() -> &'static str {
     }
 }
 
-#[exit_status::main]
-fn main() -> Result<i32> {
+fn main() {
     let args = Cli::from_args();
 
+    let bin_name = args.bin_name.clone();
+    let url = args.url.clone();
+    let native_tls = args.native_tls;
+
+    match inner_main(args) {
+        Ok(exit_code) => {
+            process::exit(exit_code);
+        }
+        Err(err) => {
+            eprintln!("{}: error: {:?}", bin_name, err);
+            if !native_tls && err.root_cause().to_string() == "invalid dnsname" {
+                eprintln!();
+                if utils::requires_native_tls(&url) {
+                    eprintln!("rustls does not support HTTPS for IP addresses.");
+                } else {
+                    // Maybe we went to https://<IP> after a redirect?
```

Comment:
We will hopefully be able to handle this case once we start using our own HTTP redirection logic https://github.com/ducaale/xh/pull/137

Reply:
Is it possible/reasonable to switch the TLS backend midway through? It'd get especially tricky if it was configured.

I think this is obscure enough that I'd rather just crash and let the user know.